addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/tenant/Tenant.java

```diff
@@ -14,10 +15,24 @@
 @JsonSerialize(as = ImmutableTenant.class)
 @Value.Style(depluralize = true)
 public interface Tenant {
+
     String getUsername();
 
     String getEmail();
 
+    /**
+     * @return The Openshift.io token used to authenticate this user in auth.openshift.io
+     */
+    TokenIdentity getIdentity();
+
     List<Namespace> getNamespaces();
 
+    @Value.Derived
+    default Namespace getDefaultUserNamespace() {
+        return getNamespaces().stream()
+                .filter(Namespace::isUserNamespace)
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException("No user namespace found for " + getUsername()));
```

Comment:
In the original code a null could be returned, don't know if that is really possible, but we might want to keep it the same?

Reply:
Afaik in the original code an exception was thrown if no namespace was found, so let's never assume this method will return null